ehitchhiking-ui/src/app/pages/main-screen/yandex-map/yandex-map.component.ts

```diff
@@ -68,43 +76,47 @@ export class YandexMapComponent implements OnInit, OnChanges {
 
 			myGeoObject.events.add('dragend', (event) => {
 				this.currentRoute.passengers[0] = {passanger: 'gleb', coordinates: this.myMark.geometry._coordinates}; // TODO:0 is bad
+        this.passengerPlaceMark.emit(false);
+        // this.myMap.geoObjects.remove(myGeoObject);
 			});
 		});
 	}
 
 	ngOnChanges(changes: SimpleChanges) {
-		if (changes.triggers && changes.triggers.currentValue) {
-			this.yandexRoutesObjects.forEach((route) => this.myMap.geoObjects.remove(route));
-		}
-		if (changes.indexRouteToDisplay) {
-			if (this.indexRouteToDisplay !== undefined && this.indexRouteToDisplay !== DELETE_ROUTE_MARKER) {
-				this.addMultiRoute(this.routes[this.indexRouteToDisplay], false);
-				this.placeMarkForPassenger();
-			}
-			if (this.indexRouteToDisplay === DELETE_ROUTE_MARKER) {
-				this.myMap.geoObjects.remove(this.yandexRoutesObjects[0]);
-				this.myMap.geoObjects.remove(this.myMark);
-			}
-		}
+    if (changes.indexRouteToDisplay && this.marker) {
+      const i = this.indexRouteToDisplay.value;
+      if (this.routes[i].displayed) {
+        this.addMultiRoute(i, false);
+      } else {
+        this.myMap.geoObjects.remove(this.routes[i].yandexRoute);
+      }
+      return;
+    }
+    if ((changes.redraw && changes.redraw.currentValue) || this.redraw === true) {
```

Comment:
changes.redraw.currentValue is equal to this.redraw

Reply:
fixed